jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java

```diff
@@ -53,20 +58,18 @@
   }
 
   private static void loadLibGmp() {
+    NativeLibrary library;
     try {
-      // Explicitly try to load the embedded version first.
-      File file = Native.extractFromResourcePath("gmp");
```

Comment:
Why not just use `Native.extractFromResourcePath("gmp", LibGmp.class.getClassLoader())` here?

As it was, the code prefers the library that is bundled as a resource over any version of the library that might otherwise be installed on the system. With your change, that is reversing since `NativeLibrary.getInstance` looks for a bundled resource only as a last resort.

Is that intentional? I'm guessing not since your problem statement is that you still want to load the bundled resource, you just need it to be class-loader-aware.